### DISCUSSION TITLE
[Feature Request] Add the hostname of executor in collection mongockChangeLog

### DISCUSSION THREAD

**Discussion by olivierboudet** [flamingock/mongock#58] [Ideas]
Sometimes, we have issues and we need to know which instance executed the changelog.
Would it be possible to have the hostname saved in the collection mongockChangeLog ?

If it is a good idea, I can try to send a PR.

**Comment by dieppa** [comment]
Hello @olivierboudet,

 I like your idea, it sounds interesting and helpful. Thanks. 
 
 However, please explain me how you are thinking to do it, so we agree before you invest your time 😄

**Reply by olivierboudet** [reply]
Yeah, I just cloned the repository. I suppose the modifications will be part of changock instead.
I am thinking to : 
- add a `executionHostname` to `ChangeEntry`
- update `io.changock.runner.core.MigrationExecutor#createChangeEntryInstance` to fetch the hostname
```
String hostname;
try {
    value = InetAddress.getLocalHost().getHostName();
} catch (Throwable e) {
    value = "unknown";
}
```
- from what I seen, the drivers should not need to be modified

**Reply by dieppa** [reply]
It looks fine to me. I would add an optional prefix for a service identifier, which would be added by configuration. Empty if not provided. Like
```yaml
mongock:
   service-identifier: client-service
```

then the expected output could be something like: `user-MacBook-Pro.local::client-service` or just`user-MacBook-Pro.local` if the service identifier is not provided.

**Comment by olivierboudet** [comment]
Hello @dieppa, I have done two branches on my repos here : 

https://github.com/olivierboudet/mongock/tree/issue-316-add-hostname
https://github.com/olivierboudet/changock/tree/issue-316-add-hostname

Do you want me to send PR ?
